src/main/java/net/neoforged/neoforge/event/entity/living/ProjectileWeaponLoadEvent.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.event.entity.living;
+
+import java.util.List;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.BowItem;
+import net.minecraft.world.item.CrossbowItem;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.ProjectileWeaponItem;
+import net.minecraft.world.level.Level;
+import net.neoforged.bus.api.ICancellableEvent;
+
+/**
+ * ProjectileWeaponLoadEvent is fired when a living entity is loading projectiles for a {@link ProjectileWeaponItem}.
+ */
+public abstract class ProjectileWeaponLoadEvent extends LivingEvent {
+    private final ItemStack weaponItem;
+    private final ItemStack projectileItem;
+
+    ProjectileWeaponLoadEvent(LivingEntity entity, ItemStack weaponItem, ItemStack projectile) {
```

Comment:
just mark this as internal for consistency please.

Reply:
Tbh I'm not sure what consistency is expected here?
Although I believe it's correct to make `Pre`/`Post` styled event to be "sealed", it's simply just not the common practice to use `@ApiStatus.Internal` on their constructors from my observation.
Other nested event classes don't use `@ApiStatus.Internal` at all on the top level constructor, `Pre`/`Post` styled event like `LivingDamageEvent` just simply uses a private constructor (that noticed me I could just use private instead of package private), event with more children like `PlayerXpEvent` even uses a public constructor (which should probably move to private instead).
If we are going to set up a general style on the event's constructors, it would belong to a clean up PR and need to be discussed in advance.